pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -634,6 +638,59 @@ func init() {
 	}
 
 	err = InfraSAs.addServiceAccount(
+		InfraPetSetControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: PetSetControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// PetSetController.podCache.ListWatch
```

Comment:
This means its not using a shared informer.  Either you wired it wrong or we need an upstream fix.